CMakeLists.txt

```diff
@@ -0,0 +1,106 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(backtrace)
+
+# Automake uses -frandom-seed initialized with file name of given file
+# but AFAIK it can't be done on CMake, so here's always same seed
+set(CMAKE_CXX_FLAGS "-DHAVE_CONFIG_H -funwind-tables -frandom-seed=mySeed -W -Wall -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wmissing-format-attribute -Wcast-qual -g -O2")
+set(CMAKE_C_FLAGS "-DHAVE_CONFIG_H -funwind-tables -frandom-seed=mySeed -W -Wall -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition -Wmissing-format-attribute -Wcast-qual -g -O2")
+
+file(GLOB sources 
+	atomic.c dwarf.c fileline.c posix.c print.c sort.c state.c backtrace.c
+	simple.c pecoff.c read.c alloc.c config.h
+)
+file(GLOB export_headers
+	${CMAKE_CURRENT_BINARY_DIR}/backtrace.h
+	${CMAKE_CURRENT_BINARY_DIR}/backtrace-supported.h
+)
+
+add_library(${PROJECT_NAME} ${sources} ${export_headers})
+include_directories(${PROJECT_NAME} ${CMAKE_CURRENT_BINARY_DIR})
+
+if(CMAKE_COMPILER_IS_GNUCC)
+	set(BACKTRACE_SUPPORTED 1)
+	
+	# Assume multi-threaded environment
+	set(BACKTRACE_SUPPORTS_THREADS 1)
+
+	# Assume ELF/DWARF, meaning that BACKTRACE_SUPPORTS_DATA is hard-coded on.

```

Comment:
This seems like an unfortunate assumption, since we have PE and XCOFF support, and we really need Mach-O support.  Do people not use cmake on those systems?  Though I suppose we could also fix this up later.

Reply:
I vote for adding things later, since I've never worked with XCOFF or Mach-O files and platforms supporting them. Therefore I can't add support to them, so I guess there's little more options than putting notice in README that CMake support is in it's infancy.

Reply:
fixed in 9f5df7e3f69e9caaca4234d7e94a5a9fc28e9c4a